test/jdk/sun/java2d/DirectX/OnScreenRenderingResizeTest/OnScreenRenderingResizeTest.java

```diff
@@ -114,27 +111,29 @@ public void run() {
             ex.printStackTrace();
         }
 
-        int maxW = gcBounds.width /2;
-        int maxH = gcBounds.height/2;
-        int minW = frame.getWidth();
-        int minH = frame.getHeight();
+        int maxW = gcBounds.width / 2;
+        int maxH = gcBounds.height / 2;
+        int minW = FRAME_W;
+        int minH = FRAME_H;
         int incW = 10, incH = 10, cnt = 0;
         Robot robot = null;
         if (!nocheck && gc.getColorModel().getPixelSize() > 8) {
             try {
                 robot = new Robot();
+                robot.setAutoDelay(100);
+                robot.mouseMove(0,0);
             } catch (AWTException ex) {
                 System.err.println("Robot creation failed, continuing.");
```

Comment:
Not sure if we should continue here as createScreenCapture is being used down below which would result in NPE
Also this try-block is for creation so I think setAutoDelay and mouseMove needs to be moved out of this block..
Also, better to use try-finally block to do frame.dispose()

Reply:
We will not throw NPE if robot is null. Because we have null check for robot when we want to use it.
This test was originally written to check for crash after producing artifacts, thats why i think we continue to drawImage onto resizing frame even if Robot fails. So its better if we keep it that way.

Also initially i used mouseMove() before we do screen capture, but this was causing considerably lag in drawing on my local mac machine. So i have kept it in initialization logic and i can see that mouse moves to (0, 0) and stays there until the test is completed. Next operation we do after this mouseMove is screenCapture. So keeping autodelay here or later should not matter.

I have moved whole logic under try block and disposing frame under finally block.

Reply:
SInce some coordinates are changed, can you please confirm the test can be used still to confirm JDK-6664068 meaning it fails before the fix and pass with it?

Reply:
I have changed the frame/image coordinates to use bounds from screen coordinates instead of using constants (128,128) for BufferedImage and (512,512) for VolatileImage, it doesn't change the way the test works.